go/vt/vttablet/tabletmanager/semisyncmonitor/monitor_test.go

```diff
@@ -0,0 +1,714 @@
+			res:     sqltypes.MakeTestResult(sqltypes.MakeTestFields("Variable_name|Value", "varchar|varchar"), "Rpl_semi_sync_source_wait_sessions|1", "Rpl_semi_sync_master_wait_sessions|1"),
+			wantErr: "Row count exceeded 1",
+		},
+		{
+			name:    "incorrect number of fields",
+			res:     sqltypes.MakeTestResult(sqltypes.MakeTestFields("Variable_name|Value|a", "varchar|varchar|int"), "Rpl_semi_sync_source_wait_sessions|1|2"),
+			wantErr: "unexpected number of rows received",
+		},
+		{
+			name: "Unblocked",
+			res:  sqltypes.MakeTestResult(sqltypes.MakeTestFields("Variable_name|Value", "varchar|varchar"), "Rpl_semi_sync_source_wait_sessions|0"),
+			want: false,
+		},
+		{
+			name: "Blocked",
+			res:  sqltypes.MakeTestResult(sqltypes.MakeTestFields("Variable_name|Value", "varchar|varchar"), "Rpl_semi_sync_source_wait_sessions|1"),
+			want: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			db, m := createFakeDBAndMonitor(t)
+			defer db.Close()
+			defer m.Close()
+			db.AddQuery(semiSyncWaitSessionsRead, tt.res)
+			got, err := m.isSemiSyncBlocked(context.Background())
+			if tt.wantErr != "" {
+				require.EqualError(t, err, tt.wantErr)
+				return
+			}
+			require.NoError(t, err)
+			require.EqualValues(t, tt.want, got)
+		})
+	}
+}
+
+func TestMonitorBindSideCarDBName(t *testing.T) {
+	tests := []struct {
+		query    string
+		expected string
+	}{
+		{
+			query:    semiSyncRecoverWrite,
+			expected: "INSERT INTO _vt.semisync_recover (ts) VALUES (NOW())",
+		},
+		{
+			query:    semiSyncRecoverClear,
+			expected: "TRUNCATE TABLE _vt.semisync_recover",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.query, func(t *testing.T) {
+			m := &Monitor{}
+			require.EqualValues(t, tt.expected, m.bindSideCarDBName(tt.query))
+		})
+	}
+}
+
+func TestMonitorClearAllData(t *testing.T) {
+	db, m := createFakeDBAndMonitor(t)
+	defer db.Close()
+	defer m.Close()
+	db.SetNeverFail(true)
+	m.clearAllData()
+	ql := db.QueryLog()
+	require.EqualValues(t, "truncate table _vt.semisync_recover", ql)
+}
+
+// TestMonitorWaitMechanism tests that the wait mechanism works as intended.
+// Setting the monitor to unblock state should unblock the waiters.
+func TestMonitorWaitMechanism(t *testing.T) {
+	db, m := createFakeDBAndMonitor(t)
+	defer db.Close()
+	defer m.Close()
+
+	// Add a waiter.
+	ch := m.addWaiter()
+	var waiterUnblocked atomic.Bool
+	go func() {
+		<-ch
+		waiterUnblocked.Store(true)
+	}()
+
+	// Ensure that the waiter is currently blocked.
+	require.False(t, waiterUnblocked.Load())
+
+	// Verify that setting again to being blocked doesn't unblock the waiter.
+	m.setIsBlocked(true)
+	require.False(t, waiterUnblocked.Load())
+	require.False(t, m.isClosed())
+	require.True(t, m.stillBlocked())
+
+	// Verify that setting we are no longer blocked, unblocks the waiter.
+	m.setIsBlocked(false)
+	require.Eventually(t, func() bool {
+		return waiterUnblocked.Load()
+	}, time.Second, time.Millisecond*100)
```

Comment:
IMO it's worth increasing the wait a bit to avoid flakiness when the CI hosts CPU is saturated. Same note elsewhere where we are only waiting for 1 second (you can go 1 second w/o any CPU time).

Reply:
Fixed